docs/tutorials/transaction-tutorial.md

```diff
@@ -4,41 +4,119 @@ title: Transactions
 As their name implies, these functions execute as [database transactions](https://en.wikipedia.org/wiki/Database_transaction).
 
 Transaction functions must be annotated with the [`@Transaction`](../api-reference/decorators#transaction) decorator and must have a [`TransactionContext`](../api-reference/contexts#transactioncontextt) as their first argument.
 As with other DBOS functions, inputs and outputs must be serializable to JSON.
 
 The [`TransactionContext`](../api-reference/contexts#transactioncontextt) provides a `.client` field you can use to transactionally interact with the database, so you don't need to worry about managing database connections.
-By default, this is a [Knex.js](./using-knex.md) client backed by a connection pool.
-We like Knex because it's lightweight and helps us write fast and type-safe queries.
-However, if you prefer a traditional ORM, we also support [TypeORM](./using-typeorm.md) and [Prisma](./using-prisma.md).
+DBOS supports [Knex.js](./using-knex.md), [TypeORM](./using-typeorm.md), and [Prisma](./using-prisma.md) clients as well as raw SQL through [knex.raw](https://knexjs.org/guide/raw.html).
+You can configure which client to use in your [`dbos-config.yaml`](../api-reference/configuration.md) file; Knex is used by default.
+
+Here's an example of a transaction function:
 
-Here's an example of a transaction function written using raw SQL (with [knex.raw](https://knexjs.org/guide/raw.html)):
+<Tabs groupId="database-clients">
+<TabItem value="knex" label="Knex">
 
 ```javascript
 export class Greetings {
   @Transaction()
-  static async InsertGreeting(ctxt: TransactionContext<Knex>, friend: string, note: string) {
-    await ctxt.client.raw('INSERT INTO greetings (name, note) VALUES (?, ?)', [friend, note]);
+  static async insertGreeting(ctxt: TransactionContext<Knex>, friend: string, note: string) {
+    await ctxt.client('greetings').insert({
+      name: friend,
+      note: note
+    });
   }
 }
 ```
 
-Here's the same function written using the Knex query builder:
+See our [Knex guide](./using-knex.md) for more information.
+
+</TabItem>
+<TabItem value="typeorm" label="TypeORM">
 
 ```javascript
+@Entity()
 export class Greetings {
+
+    @PrimaryGeneratedColumn()
+    id!: number;
+
+    @Column()
+    name!: string;
+
+    @Column()
+    note!: string;
+}
+
+@OrmEntities([Greetings])
+export class DBOSGreetings {
   @Transaction()
-  static async InsertGreeting(ctxt: TransactionContext<Knex>, friend: string, note: string) {
-    await ctxt.client('greetings').insert({
-      name: friend,
-      note: note
+  static async insertGreeting(ctxt: TransactionContext<EntityManager>, name: string, note: string) {
+    let entity = new Greetings();
+    entity.name = name;
+    entity.note = note;
+    await ctxt.client.save(entity);
+  }
+}
+```
+
+See our [TypeORM guide](./using-typeorm.md) for more information.
+
+
+</TabItem>
+<TabItem value="prisma" label="Prisma">
+
+```javascript
```

Comment:
For Prisma, I suggest we add `import { PrismaClient } from "@prisma/client"; // Use the generated client`
This indicates that the PrismaClient is the generated one.